resolwe/flow/serializers/data.py

```diff
@@ -30,18 +18,24 @@ class DataSerializer(ResolweBaseSerializer):
     input = ProjectableJSONField(required=False)
     output = ProjectableJSONField(required=False)
     descriptor = ProjectableJSONField(required=False)
-    process_slug = serializers.CharField(source='process.slug', read_only=True)
-    process_name = serializers.CharField(source='process.name', read_only=True)
-    process_type = serializers.CharField(source='process.type', read_only=True)
-    process_input_schema = ProjectableJSONField(source='process.input_schema', read_only=True)
-    process_output_schema = ProjectableJSONField(source='process.output_schema', read_only=True)
-    process = ResolweSlugRelatedField(queryset=Process.objects.all())
-    descriptor_schema = NestedDescriptorSchemaSerializer(required=False)
-    entity_names = serializers.SerializerMethodField()
-    collection_names = serializers.SerializerMethodField()
-
-    collections = serializers.SerializerMethodField()
-    entities = serializers.SerializerMethodField()
+    process = DictRelatedField(queryset=Process.objects.all(), serializer=ProcessSerializer)
```

Comment:
Process probably needs to be included in read-only fields.

Reply:
No, in update_protected fields, since it is required during creation.